### PR TITLE
Feature 146

### PR DESCRIPTION
**Pull request by Johk3r**
Fixes #
https://github.com/naorlivne/parse_it/issues/146
## Proposed Changes

  - Adding config option to override current valid None types

### CODE REVIEW

**Comment by naorlivne** [comment]
Overall looks good, there are a few small things In the review comments above but noting I doubt will be hard to change

**Review by naorlivne — CHANGES_REQUESTED** [review]

File: parse_it/parser.py [hunk 1]
```diff
@@ -274,7 +280,7 @@ def read_multiple_configuration_variables(self, config_names: list, default_valu
         return config_value_dict
 
     def read_all_configuration_variables(self, default_value: Optional[dict] = None, required: Optional[list] = None,
-                                         allowed_types: Optional[dict] = None) -> dict:
+                                         allowed_types: Optional[dict] = None, none_values: Optional[tuple] = None) -> dict:
```
[the remark this answers is not in the document]
Comment (Johk3r) [reply parse_it/parser.py:283]: This shouldn't be there, committed by accident.
removed.

File: test/test_prase_it.py [hunk 2]
```diff
@@ -363,6 +363,19 @@ def test_type_estimate_none(self):
         self.assertEqual(reply_uppercase, None)
         self.assertEqual(reply_mixed, None)
 
+    def test_type_estimate_none_custom(self):
```
[the remark this answers is not in the document]
Comment (Johk3r) [reply test/test_prase_it.py:366]: Above this method there's the one that tests the default values.
If that's not enough I can try to add something else.

File: parse_it/parser.py [hunk 3]
```diff
@@ -18,7 +18,7 @@ class ParseIt:
     def __init__(self, config_type_priority: Optional[list] = None, global_default_value: Any = None,
                  type_estimate: bool = True, recurse: bool = False, force_envvars_uppercase: bool = True,
                  config_location: Optional[str] = None, envvar_prefix: Optional[str] = None,
-                 custom_suffix_mapping: Optional[dict] = None, envvar_divider: Optional[str] = None):
+                 custom_suffix_mapping: Optional[dict] = None, envvar_divider: Optional[str] = None, none_values: Optional[set] = None):
```
[the remark this answers is not in the document]
Comment (Johk3r) [reply parse_it/parser.py]: Fixed

File: test/test_prase_it.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (naorlivne) [reply test/test_prase_it.py:366]: The idea is that if the default someone get changed the test should catch it, so there should be a run `parser` function without any custom `none_values` to make sure it assigns the correct default of `{{"", "null", "none"}}` and another time with custom values to make sure it assigns only those.

**Review by naorlivne — CHANGES_REQUESTED** [review]
Just the test of the parser function to cover the default & non default params passed missing

File: test/test_prase_it.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Johk3r) [reply test/test_prase_it.py:366]: I think this already covers the default values
`
    def test_type_estimate_none(self):
        reply_empty = estimate_type("")
        reply_lowercase = estimate_type("none")
        reply_null = estimate_type("null")
        reply_uppercase = estimate_type("NONE")
        reply_mixed = estimate_type("None")
        self.assertEqual(reply_empty, None)
        self.assertEqual(reply_lowercase, None)
        self.assertEqual(reply_null, None)
        self.assertEqual(reply_uppercase, None)
        self.assertEqual(reply_mixed, None)
`
https://github.com/naorlivne/parse_it/blob/949ede33d188e0699e6886a8560212da3989c025/test/test_prase_it.py#L354

[the remark this answers is not in the document]
Comment (naorlivne) [reply test/test_prase_it.py:366]: This covers it in the `estimate_type` function, but there's also the code in the `parse_it` class that needs testing

[the remark this answers is not in the document]
Comment (Johk3r) [reply test/test_prase_it.py:366]: Ha, ok I see, will add it. :)

[the remark this answers is not in the document]
Comment (Johk3r) [reply test/test_prase_it.py:366]: Added 2 more tests and a json test file with only None values

[the remark this answers is not in the document]
Comment (naorlivne) [reply test/test_prase_it.py:366]: The addition of `test_none_values.json` breaks other tests (mainly those revolving around the logic of what files config variables are taken from), details can be seen in https://cloud.drone.io/naorlivne/parse_it/124/1/2

[the remark this answers is not in the document]
Comment (Johk3r) [reply test/test_prase_it.py:367]: Sorry I didn't ran the tests locally before committing, should be fixed now.
Thank you.

**Comment by naorlivne** [comment]
LGTM, I'll merge this, if you want bragging rights feel free to add yourself to the [CONTRIBUTORS.md](https://github.com/naorlivne/parse_it/blob/master/CONTRIBUTORS.md) file

**Review by naorlivne — APPROVED** [review]
